packages/http/src/shared.ts

```diff
@@ -5,31 +5,187 @@ export type TActivityResponse = definitions["v1ActivityResponse"];
 export type TActivityId = TActivity["id"];
 export type TActivityStatus = TActivity["status"];
 export type TActivityType = TActivity["type"];
+export type TSignature = definitions["v1SignRawPayloadResult"];
+export type TSignedTransaction =
+  definitions["v1SignTransactionResult"]["signedTransaction"];
 
 export class TurnkeyActivityError extends Error {
-  activityId: TActivityId | null;
-  activityStatus: TActivityStatus | null;
-  activityType: TActivityType | null;
-  cause: Error | null;
+  activityId: TActivityId | undefined;
+  activityStatus: TActivityStatus | undefined;
+  activityType: TActivityType | undefined;
+  cause: Error | undefined;
 
   constructor(input: {
     message: string;
-    cause?: Error | null;
-    activityId?: TActivityId | null;
-    activityStatus?: TActivityStatus | null;
-    activityType?: TActivityType | null;
+    cause?: Error | undefined;
+    activityId?: TActivityId | undefined;
+    activityStatus?: TActivityStatus | undefined;
+    activityType?: TActivityType | undefined;
   }) {
     const { message, cause, activityId, activityStatus, activityType } = input;
     super(message);
 
     this.name = "TurnkeyActivityError";
-    this.activityId = activityId ?? null;
-    this.activityStatus = activityStatus ?? null;
-    this.activityType = activityType ?? null;
-    this.cause = cause ?? null;
+    this.activityId = activityId ?? undefined;
+    this.activityStatus = activityStatus ?? undefined;
+    this.activityType = activityType ?? undefined;
+    this.cause = cause ?? undefined;
   }
 }
 
+export class TurnkeyActivityConsensusNeededError extends Error {
+  activityId: TActivityId | undefined;
+  activityStatus: TActivityStatus | undefined;
+  activityType: TActivityType | undefined;
+  cause: Error | undefined;
+
+  constructor(input: {
+    message: string;
+    cause?: Error | undefined;
+    activityId?: TActivityId | undefined;
+    activityStatus?: TActivityStatus | undefined;
+    activityType?: TActivityType | undefined;
+  }) {
+    const { message, cause, activityId, activityStatus, activityType } = input;
+    super(message);
+
+    this.name = "TurnkeyActivityConsensusNeededError";
+    this.activityId = activityId ?? undefined;
+    this.activityStatus = activityStatus ?? undefined;
+    this.activityType = activityType ?? undefined;
+    this.cause = cause ?? undefined;
+  }
+}
+
+export function checkActivityStatus(input: {
```

Comment:
`checkActivityStatus` implies it's a stateless innocuous function but I think this is more heavy than it looks: it will throw errors! So maybe `assertActivityCompleted` since that's kind of what it boils down to?

Reply:
def more accurate1